src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -320,6 +322,120 @@ namespace AppInstaller::Repository
         THROW_HR(APPINSTALLER_CLI_ERROR_INVALID_SOURCE_TYPE);
     }
 
+    SourceTrustLevel ConvertToSourceTrustLevelEnum(std::string_view trustLevel)
+    {
+        std::string lowerTrustLevel = Utility::ToLower(trustLevel);
+
+        if (lowerTrustLevel == "storeorigin")
+        {
+            return SourceTrustLevel::StoreOrigin;
+        }
+        else if (lowerTrustLevel == "trusted")
+        {
+            return SourceTrustLevel::Trusted;
+        }
+        else if (lowerTrustLevel == "none")
+        {
+            return SourceTrustLevel::None;
+        }
+        else
+        {
+            THROW_HR(E_UNEXPECTED);
+        }
+    }
+
+    std::string SourceTrustLevelToString(SourceTrustLevel trustLevel)
+    {
+        switch (trustLevel)
+        {
+        case SourceTrustLevel::StoreOrigin:
+            return "StoreOrigin";
+        case SourceTrustLevel::Trusted:
+            return "Trusted";
+        case SourceTrustLevel::None:
+            return "None";
+        }
+
+        return "Unknown";
+    }
+
+    SourceTrustLevel ConvertToSourceTrustLevelEnum(std::vector<std::string> trustLevels)
+    {
+        Repository::SourceTrustLevel result = Repository::SourceTrustLevel::None;
+        for (auto& trustLevel : trustLevels)
+        {
+            Repository::SourceTrustLevel trustLevelEnum = ConvertToSourceTrustLevelEnum(trustLevel);
+            if (trustLevelEnum == Repository::SourceTrustLevel::None)
+            {
+                return Repository::SourceTrustLevel::None;
+            }
+            else if (trustLevelEnum == Repository::SourceTrustLevel::Trusted)
+            {
+                WI_SetFlag(result, Repository::SourceTrustLevel::Trusted);
+            }
+            else if (trustLevelEnum == Repository::SourceTrustLevel::StoreOrigin)
+            {
+                WI_SetFlag(result, Repository::SourceTrustLevel::StoreOrigin);
+            }
+            else
+            {
+                THROW_HR_MSG(E_UNEXPECTED, "Invalid source trust level.");
+            }
+        }
+
+        return result;
+    }
+
+    std::vector<std::string> SourceTrustLevelToList(SourceTrustLevel trustLevel)
+    {
+        if (trustLevel == Repository::SourceTrustLevel::None)
+        {
+            return { SourceTrustLevelToString(Repository::SourceTrustLevel::None) };

```

Comment:
It seems to me that it makes more sense to return an empty vector, which would mean just removing this entire half od the if/else.